### PR TITLE
PUBDEV-3363: R binding for MOJO

### PR DESCRIPTION
**Pull request by navdeep-G**
This PR provides an R binding for the new MOJO models. Simple runit is applied with this PR, as well.

---
This change is [Reviewable](https://reviewable.io/reviews/h2oai/h2o-3/219)

### CODE REVIEW

File: h2o-r/h2o-package/R/kvstore.R [hunk 1]
```diff
@@ -266,3 +266,59 @@ h2o.download_pojo <- function(model, path="", getjar=NULL, get_jar=TRUE) {
...
 }
+
+#'
+#' Download the model in MOJO format.
+#'
+#' @param model An H2OModel
+#' @param path The path where MOJO file should be saved. Saved to current directory by default.
+#' @param get_genmodel_jar If TRUE, then also download h2o-genmodel.jar and store it in folder ``path``.
+#' @return If path is "", then pretty print the MOJO to the console.
+#'         Otherwise save it to the specified directory.
```
Comment (st-pasha) [inline h2o-r/h2o-package/R/kvstore.R]: I don't think you can pretty-print a zip file to console...

Comment (st-pasha) [inline h2o-r/h2o-package/R/kvstore.R]: In Python return value is the name of the mojo file created.

File: h2o-r/h2o-package/R/kvstore.R [hunk 2]
```diff
@@ -266,3 +266,59 @@ h2o.download_pojo <- function(model, path="", getjar=NULL, get_jar=TRUE) {
...
+    tmp = tryCatch(getBinaryURL(url = url,
+                        useragent = R.version.string),
+                 error = function(x) { .__curlError <<- TRUE; .__curlErrorMessage <<- x$message })
+    if (! .__curlError) {
+      jar.path <- paste0(path, "/h2o-genmodel.jar")
+      writeBin(tmp, jar.path, useBytes = TRUE)
+    }
+  }
+
+  print( paste0("MOJO written to: ", mojo.path) )
```
Comment (st-pasha) [inline h2o-r/h2o-package/R/kvstore.R]: I don't see where you return the file name here...

Comment (navdeep-G) [inline h2o-r/h2o-package/R/kvstore.R]: The path contains the file name,i.e, `GBM_model_R_... .zip`

Comment (st-pasha) [inline h2o-r/h2o-package/R/kvstore.R]: I may be misunderstanding R, but it seems to me that the function prints and then returns string "MOJO written to: GBM_model_R_12345.zip", which contains the file name but IS NOT the filename (as your @return annotation in the function doc states).

On the Python side I was making an assert of the form
```
mojo_file = model.download_mojo(path)
assert os.path.exists(mojo_file)
```

Comment (navdeep-G) [inline h2o-r/h2o-package/R/kvstore.R]: Thats probably an old commit. I return the following now:
`print( paste0("MOJO file name: ", model_id,".zip") )
 print( paste0("MOJO written to: ", mojo.path) )`

**Pull request merged**